src/redux/epics/userInputEpic.js

```diff
@@ -0,0 +1,15 @@
+import { DEBOUNCE_TIME } from '../../shared/constants'
+import { actions as dataSliceActions } from '../data/dataSlice'
+import { debounceTime, filter, mapTo } from 'rxjs/operators'
+import { updateUserInput } from '../userInput/userInputSlice'
+
+const userInputEpic = (action$) => {
+  return action$.pipe(
+    debounceTime(DEBOUNCE_TIME),
```

Comment:
il debonce andrebbe fatto dopo il filter in modo da non dover avere problemi se si hanno troppe azioni